tensorpack/dataflow/base.py

```diff
@@ -39,29 +40,59 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return False
 
 
-@six.add_metaclass(ABCMeta)
+class DataFlowMeta(type):
+    """
+    DataFlow uses "__iter__()" and "__len__()" instead of
+    "get_data()" and "size()". This add back-compatibility.
+    """
+    def __new__(meta, name, bases, dct):
+        # "'get_data' in dct" is important as we just care about DataFlow itself
+        # and not derived classes.
+        if '__iter__' not in dct and 'get_data' in dct:
+            dct['__iter__'] = dct['get_data']
+            log_deprecated("DataFlow.get_data()",
+                           "use DataFlow.__iter__() instead!",
+                           "2018-12-30")
+        if '__len__' not in dct and 'size' in dct:
+            dct['__len__'] = dct['size']
+            log_deprecated("DataFlow.size()",
+                           "use DataFlow.__len__() instead!",
+                           "2018-12-30")
+        return super(DataFlowMeta, meta).__new__(meta, name, bases, dct)
+
+    def __init__(cls, name, bases, dct):
+        super(DataFlowMeta, cls).__init__(name, bases, dct)
+
+
+@six.add_metaclass(DataFlowMeta)
 class DataFlow(object):
     """ Base class for all DataFlow """
 
     @abstractmethod
```

Comment:
When the metaclass is not ABCMeta I guess `abstractmethod` will have no effect.
Instead we should check in our own meta, that one of the two methods `get_data` and `__iter__` was implemented.

Reply:
I just use `ABCMeta` as the parent and `@abstractmethod` is working